DependencyInjection/Compiler/ConfigurationCheckPass.php

```diff
@@ -30,7 +30,7 @@ public function process(ContainerBuilder $container)
         }
 
         if ($container->has('fos_rest.view_response_listener') && isset($container->getParameter('kernel.bundles')['SensioFrameworkExtraBundle'])) {
-            if (!$container->has('sensio_framework_extra.view.listener')) {
+            if (!($container->has('sensio_framework_extra.view.listener') ^ $container->hasDefinition('Sensio\Bundle\FrameworkExtraBundle\EventListener\TemplateListener'))) {
```

Comment:
Using a bitwise xor operator (`^`) here seems odd. Also, I don't see why the case that both definitions exist should be excluded explicitly. I'd prefer `||` instead.

Also, since we have php 5.5 here, you can safely use `TemplateListener::class` instead of the FQCN string.

Reply:
If SensioFrameworkExtraBundle is not installed (and isnt required by FOSRestBundle), you cannot use TemplateListener::class because you will end up with the string 'TemplateListener' which is not the same thing.

Reply:
@merk This is wrong. `::class` will always resolve to the FQCN in your `use` statement, no matter if that class exists or not. Also, neither `use` nor `::class` will trigger autoloading. So it is indeed perfectly save to use it here.

https://3v4l.org/RkQNN